added storage and memory parts: dram, ssd, sd card reader, rom

rsnancollas opened the pull request.

rsnancollas committed:
added storage and memory parts: dram, ssd, sd card reader, rom

asadate commented on part-spec/storage_memory.json:
```diff
@@ -0,0 +1,181 @@
+                        "mini",
+                        "full size"
+                    ],
+                    "type": "string"
```
Add capacity, speed/data rate properties just like the above

asadate commented on part-spec/storage_memory.json:
```diff
@@ -0,0 +1,181 @@
+                "interface": {
+                    "description": "interface of ssd to host",
+                    "examples": [
+                        "emmc 5.1",
```
remove emmc 5.1 from the examples since this is not an SSD

asadate commented on part-spec/storage_memory.json:
```diff
@@ -0,0 +1,181 @@
+            }
+        },
+        {
+            "title": "sdCardReader",
```
change title to SD card

asadate commented on part-spec/storage_memory.json:
```diff
@@ -0,0 +1,181 @@
+            "title": "ssd",
+            "type": "object",
+            "required": [
+                "componentID"
```
Add capacity as a required properties for all storage/memory

asadate commented on part-spec/storage_memory.json:
```diff
@@ -0,0 +1,181 @@
+                "type": {
+                    "description": "type of ssd storage as defined by interface and technology",
+                    "examples": [
+                        "emmc",
```
remove emmc from examples

asadate committed:
Update storage_memory.json

Made following updates: added "capacity" and a required property; remove emmc from ssd interface examples; changed "sdCard reader" to "sd card"; added "capacity" and "datarate" property for the sd card;

asadate reviewed (approved):
Approving changes to bring these specs up to date with spec rev0.5.
Reviews for properties are still to be done later.
